### PR TITLE
Add support for forward type zones and zone type auto-detection

### PR DESCRIPTION
**Pull request by GR360RY**
* Added support for forward zone types and per zone forwarders.
* Added support for zone type auto-detection.
* Added ability to specify zone type explicitly to override auto-detection. This is useful when testing dns infrastructure changes in staging environment and primary server address doesn't much zone configuration.
* Added support for group bind_zones inventory that allows to have a single file with all zones definitions for all DNS servers.
* Added shared_inventory molecule scenario to test the setup
* Updated documentation
* Removed `.travis.yml`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -185,6 +195,48 @@ Remark that only the network part should be specified here! When specifying a cl
...
+* When `primaries` record is updated to switch primary to secondary server roles, zones will be wiped out and recreated from template as we yet to support dynamic updates for existing zones.
+---
+
+Zone types can be also defined explicity in per host inventory to skip autodetection:
+
+```Yaml
+# Primary Server
+bind_zones:
+  - name: mydomain.com
+    type: master
```
Comment (bertvv) [inline README.md:221]: Should be `primary`

File: README.md [hunk 2]
```diff
@@ -185,6 +195,48 @@ Remark that only the network part should be specified here! When specifying a cl
...
+    primaries:
+      - 192.0.2.1
+...
+# Secondary Server
+bind_zones:
+  - name: mydomain.com
+      type: slave
```
Comment (bertvv) [inline README.md:228]: Should be `secondary`

**Comment by bertvv** [comment]
Excellent work! Thank you so much for this contribution!

If you can fix the typos in the documentation, we're good to go & I will release a new version based on this addition.

**Comment by GR360RY** [comment]
@bertvv , just forced pushed the doc update to address the typos. I think we are good to go.
Working on additional changes al well, so new pull requests are coming :)

**Comment by bertvv** [comment]
Cool, thanks! I'll postpone drafting a new release then. I'm also going to give you access to the repo so you can merge PRs yourself.

**Comment by GR360RY** [comment]
@bertvv , thanks for the maintainer level access to the repo! Will it be o.k by you if I'll merge the PR following the completion of CI tests ?

File: README.md [hunk 3]
```diff
@@ -185,6 +195,48 @@ Remark that only the network part should be specified here! When specifying a cl
 
 Based on the idea and examples detailed at <https://linuxmonk.ch/wordpress/index.php/2016/managing-dns-zones-with-ansible/> for the gdnsd package, the zone files are fully idempotent, and thus only get updated if "real" content changes.
 
+### Zone types and Zone type auto-detection
+
+Zone `type` is an optional zone parameter that defines if the zone type should be of `primary`, `secondary` or `forward` type. When `type` parameter is ommited, zone type will be autodetected based on the intersection of host IP addresses and `primaries` record when configuring primary or secondary zone. When `primaries` is not defined and `forwarders` is defined, the zone type will be set to `forward`.
+
+Zone autodetection functionality is especially usefull when deploying multi-site DNS infrastrucutre. It is convinient to have a "shared" `bind_zones` definitions in a single group inventory file for all dns servers ( ex. `group_vars\dns.yml`). Such an approach allows to switch between primary and secondary server(s) roles by updating `primaries` record only and rerunning the playbook.
```
Comment (blofeldthefish) [inline README.md:202]: Hi Gregory,  I have just returned from a break, to a flurry of emails, and some awesome work by yourself to streamline this role.  So I'd also like to offer thanks in this regard...

I noted a couple of spelling corrections in the README, 
convinient - convenient
omited - omitted

Comment (GR360RY) [reply README.md:202]: Go it. I will make sure to add spell checker plugin to my editor :). I will address all spelling mistakes shortly.

**Comment by bertvv** [comment]
Yes, you can merge this PR when it's ready!

**Pull request merged**